### PR TITLE
[11.0][FIX] Website Sale Time Frame Automatic Open/Close

### PR DESCRIPTION
**Pull request by robinkeunen**
- wrong type for datetimes
- used dates instead of datetime
- localize dates for cron job

### CODE REVIEW

**Comment by houssine78** [comment]
@robinkeunen il y a une raison pour laquelle cela n'a jamais été mergé ?

**Comment by robinkeunen** [comment]
@houssine78 No idea, it's not in the gitaggretor configs for 11-test or 11-prod so you can close it by me.

**Comment by houssine78** [comment]
@robinkeunen ok but it was kind of a fix and I see that there is some things for time zone. So it worth to be sure that we don't need it.

maybe that @vdewulf would remember ? :)

**Comment by houssine78** [comment]
@robinkeunen à priori c'est lié à cette tache là qui est en backlog.

https://gestion.coopiteasy.be/web#id=3220&view_type=form&model=project.task&action=479

**Comment by vdewulf** [comment]
@houssine78 @robinkeunen indeed, a fix on a functionnality that is not used by our 2 customers on our infra. But might be useful for cath's customer. To check with her and keep this code. I think we didn't finish because I found that the fix was not totally correct :)

**Comment by mclaeysb** [comment]
I did a rebase on `11.0` (where I fixed the tests first), fixed the tests and we are now testing this functionally. After that I am proposing to merge this.

**Comment by mclaeysb** [comment]
Thanks @robinkeunen for your tips on catching the error.
Thanks @houssine78 for your tips on sending the emails.

Take a look if you want to!

File: distribution_circuits_sale/models/time_frame.py
Comment (houssine78) [inline distribution_circuits_sale/models/time_frame.py]: did you test it being logged with the supervisor as a normal user and not the administrator ? cause in that case it could not receive the email. a solution could be to use sudo() to force the mail being sent by the admin

File: distribution_circuits_sale/models/time_frame.py
Comment (houssine78) [inline distribution_circuits_sale/models/time_frame.py]: same here

File: website_sale_preset_carts/models/time_frame.py
Comment (houssine78) [inline website_sale_preset_carts/models/time_frame.py:140]: frame_subscription_id.id could not be False. and if frame_subscription_id is false frame_subscription_id.id will raise an error.

So not sure ('subscription_id', '!=', False) is necessary

File: distribution_circuits_sale/__manifest__.py
Comment (houssine78) [inline distribution_circuits_sale/__manifest__.py]: website : https://www.coopiteasy.be and remove duplicate
increment version
add you name in the header of all the py files

**Review by houssine78 — COMMENTED** [review]
could you give a look at my cooments and suggestion ?

File: website_sale_preset_carts/models/time_frame.py
Comment (mclaeysb) [reply website_sale_preset_carts/models/time_frame.py:140]: See #25

File: distribution_circuits_sale/models/time_frame.py
Comment (mclaeysb) [reply distribution_circuits_sale/models/time_frame.py]: The demo user does not have access to EMH, so I could not easily test this. Sending using the administrator is fine actually, and since I have to go forward with this I implemented your suggestion of `sudo()` for safety, and added an extra test on the mail.

**Comment by mclaeysb** [comment]
Some context: waiting for #23 and then #25 to merge, and feedback from the client.
In 2f45044 I refactored to put this entire functionality in a separate module!

**Comment by mclaeysb** [comment]
Positive feedback from client! Ready to merge - take into account the **deploy notes**!

**Comment by houssine78** [comment]
Thanks for the feedback @mclaeysb 

this should go to the corresponding task. could you tag me on it please ?

**Comment by mclaeysb** [comment]
The deploy notes I'm referencing are those I added to the task last week :) I just left a note here to give an update since my note from 24/6.

**Review by houssine78 — APPROVED** [review]

**Pull request merged**
